### PR TITLE
feat(discover): Adding aggregates to the autocomplete

### PR DESCRIPTION
**Pull request by wmak**
- Including the default params to the autocomplete to make them easier
  to use. The idea being that `percentile(transaction.duration, 0.5):`
  is easier to update than simply `percentile():`
- This also fixes an issue where `apdex(400):0` would be interpreted
  `apdex(300)(400):0` instead, since the old function `apdex` would be
  found in the query and replaced

### CODE REVIEW

File: src/sentry/static/sentry/app/views/events/searchBar.jsx [hunk 1]
```diff
@@ -24,6 +24,15 @@ const SEARCH_SPECIAL_CHARS_REGEXP = new RegExp(
 const FIELD_TAGS = Object.fromEntries(
   Object.keys(FIELDS).map(item => [item, {key: item, name: item}])
 );
+function create_aggregation(item) {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/events/searchBar.jsx:27]:
```suggestion
function createAggregation(item) {
```

File: src/sentry/static/sentry/app/views/events/searchBar.jsx [hunk 2]
```diff
@@ -24,6 +24,15 @@ const SEARCH_SPECIAL_CHARS_REGEXP = new RegExp(
...
+  const aggregation = `${item}(${AGGREGATIONS[item].parameters
+    .map(param => param?.defaultValue)
+    .join()})`;
+  return [aggregation, {key: aggregation, name: aggregation}];
+}
+const FUNCTION_FIELDS = Object.fromEntries(
+  Object.keys(AGGREGATIONS).map(item => create_aggregation(item))
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/events/searchBar.jsx:34]:
```suggestion
  Object.keys(AGGREGATIONS).map(item => createAggregation(item))
```

File: src/sentry/static/sentry/app/views/events/searchBar.jsx [hunk 3]
```diff
@@ -24,6 +24,15 @@ const SEARCH_SPECIAL_CHARS_REGEXP = new RegExp(
 const FIELD_TAGS = Object.fromEntries(
   Object.keys(FIELDS).map(item => [item, {key: item, name: item}])
 );
+function create_aggregation(item) {
+  const aggregation = `${item}(${AGGREGATIONS[item].parameters
+    .map(param => param?.defaultValue)
+    .join()})`;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/events/searchBar.jsx:30]: Won't this not include `,` between the arguments of `percentile()`? Should we use the currently selected fields as inputs to the function list? Right now it seems like a user could select apdex(800) and the autocomplete would suggest apdex(300) which isn't ideal.

Comment (wmak) [reply src/sentry/static/sentry/app/views/events/searchBar.jsx:30]: `.join()` defaults to `,` But I'll be more explicit in the future and include it.

Also that's a really good point, I'll update this to use selected columns instead :+1:

**Review by evanh — APPROVED** [review]

File: src/sentry/static/sentry/app/views/events/searchBar.jsx (hunk 3 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/events/searchBar.jsx]: Neat, I didn't know that Array.join() defaulted to , :+1:

File: src/sentry/static/sentry/app/views/events/searchBar.jsx [hunk 4]
```diff
@@ -78,12 +79,20 @@ class SearchBar extends React.PureComponent {
   prepareQuery = query => query.replace(SEARCH_SPECIAL_CHARS_REGEXP, '');
 
   getTagList() {
-    const {organization, tags, omitTags} = this.props;
-    const fields = organization.features.includes('transaction-events')
-      ? FIELD_TAGS
+    const {fields, organization, tags, omitTags} = this.props;
+    const function_tags = fields
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/events/searchBar.jsx:83]:
```suggestion
    const functionTags = fields
```

File: src/sentry/static/sentry/app/views/events/searchBar.jsx [hunk 5]
```diff
@@ -78,12 +79,20 @@ class SearchBar extends React.PureComponent {
...
+      ? Object.fromEntries(
+          fields
+            .filter(item => !Object.keys(FIELD_TAGS).includes(item.field))
+            .map(item => [item.field, {key: item.field, name: item.field}])
+        )
+      : {};
+
+    const field_tags = organization.features.includes('transaction-events')
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/events/searchBar.jsx:91]:
```suggestion
    const fieldTags = organization.features.includes('transaction-events')
```

File: src/sentry/static/sentry/app/views/events/searchBar.jsx [hunk 6]
```diff
@@ -78,12 +79,20 @@ class SearchBar extends React.PureComponent {
...
+      ? assign(FIELD_TAGS, function_tags)
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/events/searchBar.jsx:92]:
```suggestion
      ? assign(FIELD_TAGS, functionTags)
```

File: src/sentry/static/sentry/app/views/events/searchBar.jsx [hunk 7]
```diff
@@ -78,12 +79,20 @@ class SearchBar extends React.PureComponent {
...
       : omit(FIELD_TAGS, TRACING_FIELDS);
 
-    const combined = assign({}, tags, fields);
+    const combined = assign({}, tags, field_tags);
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/events/searchBar.jsx:95]:
```suggestion
    const combined = assign({}, tags, fieldTags);
```

**Review by markstory — APPROVED** [review]
Few small naming/style nits.

**Review by dashed — APPROVED** [review]

**Pull request merged**